projects/components/src/filtering/filter-bar/filter-bar.component.ts

```diff
@@ -175,10 +193,10 @@ export class FilterBarComponent implements OnChanges, OnInit, OnDestroy {
   }
 
   private updateFilter(oldFilter: Filter, newFilter: Filter): void {
-    this.onFiltersChanged(this.filterBarService.updateFilter(this.internalFiltersSubject$.value, oldFilter, newFilter));
+    this.onFiltersChanged(this.filterBarService.updateFilter(this.filters || [], oldFilter, newFilter));
```

Comment:
When do we use the subject's value and when do we use this.filters? We seem to use different ways in different places which makes it very confusion to understand as a reader.